### PR TITLE
Deprecated binding for 1.7

### LINKED ISSUE #10043: Delete /binding
Forked from https://github.com/GoogleCloudPlatform/kubernetes/pull/9991#discussion_r32697587

We need to update the scheduler (ours and mesos) to use pods/binding instead of /binding and then we can delete /binding.

### PR DESCRIPTION
**Pull request by k82cn**
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #10043

```release-note
Deprecated Binding objects in 1.7.
```

### CODE REVIEW

**Comment by k82cn** [comment]
hi team, would you help to review it? I'd like to get it merge in 1.7 :).

File: pkg/api/types.go [hunk 1]
```diff
@@ -3135,7 +3135,8 @@ type NamespaceList struct {
 	Items []Namespace
 }
 
-// Binding ties one object to another - for example, a pod is bound to a node by a scheduler.
+// Binding ties one object to another; for example, a pod is bound to a node by a scheduler.
+// Deprecated in 1.7.
```
Comment (gyliu513) [inline pkg/api/types.go]: How about move `Deprecated in 1.7.` before the comments?

Such as 

```
// Deprecated in 1.7.
// Binding ties one object to another; for example, a pod is bound to a node by a scheduler.
```

File: pkg/api/v1/types.go [hunk 2]
```diff
@@ -3593,8 +3593,8 @@ type NamespaceList struct {
 	Items []Namespace `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
 
-// Binding ties one object to another.
-// For example, a pod is bound to a node by a scheduler.
+// Binding ties one object to another; for example, a pod is bound to a node by a scheduler.
+// Deprecated in 1.7.
```
Comment (gyliu513) [inline pkg/api/v1/types.go]: Ditto

**Comment by smarterclayton** [comment]
@kubernetes/sig-api-machinery-pr-reviews

File: pkg/api/types.go (hunk 1 above)
Comment (k82cn) [inline pkg/api/types.go]: seems both style are there in the doc :).

Comment (gyliu513) [inline pkg/api/types.go]: Never mind, depend on you.

**Comment by k82cn** [comment]
rebased :).

**Comment by sttts** [comment]
Protobuf is broken. Otherwise lgtm.

**Comment by k82cn** [comment]
Green now :) 

@ sttts , @deads2k can you help to approve? It's better to deprecate it as early as possible :).

**Comment by sttts** [comment]
@smarterclayton approved?

**Comment by k82cn** [comment]
@smarterclayton , would you help to approve this? This PR deprecated "Binding" (`/binding`) as we discuss. It's better to deprecate in this release (1.7) :).

**Comment by smarterclayton** [comment]
/approve

Starts the clock on removing the old resource

**Comment by smarterclayton** [comment]
No release risk, communicates intent.

**Comment by smarterclayton** [comment]
Actually, one tweak?  Can you say expand the text to say "deprecated in 1.7, please use the bindings subresource of pods instead."?

**Comment by k82cn** [comment]
sure, np; updates it right now :).

**Comment by k82cn** [comment]
Done, ping for lgtm :).

**Pull request merged**